src/theory/bv/bitblast/proof_bitblaster.cpp

```diff
@@ -23,6 +23,69 @@ namespace cvc5 {
 namespace theory {
 namespace bv {
 
+BitblastProofGenerator::BitblastProofGenerator(ProofNodeManager* pnm,
+                                               TConvProofGenerator* tcpg)
+    : d_pnm(pnm), d_tcpg(tcpg)
+{
+}
+
+std::shared_ptr<ProofNode> BitblastProofGenerator::getProofFor(Node eq)
+{
+  const auto& [t, bbt] = d_cache.at(eq);
+
+  CDProof cdp(d_pnm);
+  /* Coarse-grained bit-blast step. */
+  if (t.isNull())
+  {
+    cdp.addStep(eq, PfRule::BV_BITBLAST, {}, {eq});
+  }
+  else
+  {
+    /* Fine-grained bit-blast step for bit-blasting bit-vector atoms (bbAtom()).
+     * Bit-blasting atoms involves three steps:
+     * 1) pre-rewrite: rewrite atom
+     * 2) bit-blast rewritten atom
+     * 3) post-rewrite: rewrite bit-blasted atom
+     * bit-blasts and post-rewrites.
+     */
+
+    Node rwt = Rewriter::rewrite(t);
+
+    std::vector<Node> transSteps;
+
+    // Record pre-rewrite of bit-blasted term.
+    if (t != rwt)
+    {
+      cdp.addStep(t.eqNode(rwt), PfRule::REWRITE, {}, {t});
+      transSteps.push_back(t.eqNode(rwt));
+    }
+
+    // Record bit-blast step.
+    cdp.addProof(d_tcpg->getProofFor(rwt.eqNode(bbt)));
+    transSteps.push_back(rwt.eqNode(bbt));
+
+    // Record post-rewrite of bit-blasted term.
+    Node rwbbt = Rewriter::rewrite(bbt);
+    if (bbt != rwbbt)
+    {
+      cdp.addStep(bbt.eqNode(rwbbt), PfRule::REWRITE, {}, {bbt});
+      transSteps.push_back(bbt.eqNode(rwbbt));
+    }
+
+    if (transSteps.size() > 1)
```

Comment:
reforce here that the `else` of this case is a no-op because `eq` will already have been justified above